Web Inspector: Elements tab DOM tree view reduces deeply-nested nodes to one character wide

rcaliman-apple opened the pull request:
#### c0cef41981cc6b72b407b35d20183df01caeebf5
<pre>
Web Inspector: Elements tab DOM tree view reduces deeply-nested nodes to one character wide
<a href="https://bugs.webkit.org/show_bug.cgi?id=234376">https://bugs.webkit.org/show_bug.cgi?id=234376</a>
<a href="https://rdar.apple.com/86833831">rdar://86833831</a>

Reviewed by BJ Burg.

Set a minimum width for DOM tree outline items to prevent wrapping to unusable widths.

Ensure the containing block for the absolutely-positioned selection area
(the blue/gray highlight for selected tree items) spans the full width of the scroll container.

Without this change, the absolutely-positioned element spans only the width of the scroll port,
and appears cut-off when scrolling horizontally.

* Source/WebInspectorUI/UserInterface/Views/DOMTreeContentView.css:
(.content-view.dom-tree &gt; .tree-outline-wrapper):
* Source/WebInspectorUI/UserInterface/Views/DOMTreeContentView.js:
(WI.DOMTreeContentView):
* Source/WebInspectorUI/UserInterface/Views/DOMTreeOutline.css:
(.tree-outline.dom):
(.tree-outline.dom li):
* Source/WebInspectorUI/UserInterface/Views/TreeOutline.js:
(WI.TreeOutline.prototype.treeElementFromEvent):

Canonical link: <a href="https://commits.webkit.org/286163@main">https://commits.webkit.org/286163@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9ec7967ec08702545aa5c8ec65bcf494734a1598

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/74185 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/53614 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/26996 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/78559 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/25423 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/62747 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/1399 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/58318 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/16659 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/77252 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/48480 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/63815 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/38728 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/45403 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/21311 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/23756 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/66864 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/21657 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/80078 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/1502 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/844 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/66616 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/1647 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/63832 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/65889 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16516 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/9831 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/7988 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/1466 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/1495 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/1483 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/1502 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

patrickangle commented on Source/WebInspectorUI/UserInterface/Views/TreeOutline.js:
```diff
@@ -859,7 +859,7 @@ WI.TreeOutline = class TreeOutline extends WI.Object
         // (and partially hidden), in which case we use the edge of its container.
 
-        let scrollContainer = this.element.parentElement;
+        let scrollContainer = this.element.parentElement.parentElement;
```
Are we sure this is safe to change here in such a generic class? You've only changed how a TreeOutline is contained by DOMTreeContentView, but many other places in Web Inspector use this class.

rcaliman-apple replied:
Discussed with Patrick offline: I'll attempt to move the wrapper creation inside `WI.DOMTreeOutline` and expose a getter for the appropriate scroll container depending on context for `WI.TreeOutline` / `WI.DOMTreeOutline` instead of relying on direct `parentElement` calls.

rcaliman-apple replied:
> Are we sure this is safe to change here in such a generic class? You've only changed how a TreeOutline is contained by DOMTreeContentView, but many other places in Web Inspector use this class.

@patrickangle Turns out I could use `width:fit-content` on the existing containing block, the  top-level `<ol>` created by `WI.DOMTreeOutline`, without introducing another wrapper or changing the scroll container.

I did an inspection of other uses of `WI.DOMTreeOutline` in Web Inspector and there is no visible impact elsewhere.

I did find an existing issue where their containers inhibit horizontal scroll and that prevents full expanding deeply nested node trees. At least one of them isn't a straightforward change and I don't want to block this patch on it. I filed https://bugs.webkit.org/show_bug.cgi?id=282348 as a separate issue.

rcaliman-apple committed:
Web Inspector: Elements tab DOM tree view reduces deeply-nested nodes to one character wide
https://bugs.webkit.org/show_bug.cgi?id=234376
rdar://86833831

Reviewed by BJ Burg.

Set a minimum width for DOM tree outline items to prevent wrapping to unusable widths.

Ensure the containing block for the absolutely-positioned selection area
(the blue/gray highlight for selected tree items) spans the full width of the scroll container.

Without this change, the absolutely-positioned element spans only the width of the scroll port,
and appears cut-off when scrolling horizontally.

* Source/WebInspectorUI/UserInterface/Views/DOMTreeContentView.css:
(.content-view.dom-tree > .tree-outline-wrapper):
* Source/WebInspectorUI/UserInterface/Views/DOMTreeContentView.js:
(WI.DOMTreeContentView):
* Source/WebInspectorUI/UserInterface/Views/DOMTreeOutline.css:
(.tree-outline.dom):
(.tree-outline.dom li):
* Source/WebInspectorUI/UserInterface/Views/TreeOutline.js:
(WI.TreeOutline.prototype.treeElementFromEvent):

Canonical link: https://commits.webkit.org/286163@main
